[qmc5883l] Added drdy_pin option to allow it to run max rate (#5417)
OttoWinter/esphomedocs: content/components/sensor/qmc5883l.md

  temperature value is accurate". Above example has an offset that you need to calibrate. All options from
  [Sensor](#config-sensor).

- **range** (*Optional*, enum): The range parameter for the sensor.
- **oversampling** (*Optional*, enum): The oversampling parameter for the sensor.
- **update_interval** (*Optional*, [Time](#config-time)): The interval to check the sensor. To acheive the highest
  date rate of `200Hz` this can be set to zero when `drdy_pin` is configured. Defaults to `60s`.
- **id** (*Optional*, [ID](#config-id)): Manually specify the ID used for code generation.
- **drdy_pin** (*Optional*, [Pin](#config-pin)): The data ready pin. It is recommended to connect the data
  ready pin when using a low `update_interval`.
- **data_rate** (*Optional*, enum): The data rate parameter for the sensor.

## Data Rate Options

By default, the QMC5883L data rate is 200Hz. You can, however,
configure this amount. Possible values:

- `200Hz` (default)
- `100Hz`
- `50Hz`
- `10Hz`

> > NOTE: To achieve the full data rate as set by the `data_rate` value, it is *highly advised* to set the `update_interval` to `0ms`.

Reviewer: ```suggestion
> [!NOTE]
> To achieve the full data rate as set by the `data_rate` value, it is highly advised to set the `update_interval` to zero.
```

  temperature value is accurate". Above example has an offset that you need to calibrate. All options from
  [Sensor](#config-sensor).

- **range** (*Optional*, enum): The range parameter for the sensor.
- **oversampling** (*Optional*, enum): The oversampling parameter for the sensor.
- **update_interval** (*Optional*, [Time](#config-time)): The interval to check the sensor. To acheive the highest
  date rate of `200Hz` this can be set to zero when `drdy_pin` is configured. Defaults to `60s`.
- **id** (*Optional*, [ID](#config-id)): Manually specify the ID used for code generation.
- **drdy_pin** (*Optional*, [Pin](#config-pin)): The data ready pin. It is recommended to connect the data
  ready pin when using a low `update_interval`.
- **data_rate** (*Optional*, enum): The data rate parameter for the sensor.

## Data Rate Options

By default, the QMC5883L data rate is 200Hz. You can, however,
configure this amount. Possible values:

- `200Hz` (default)
- `100Hz`
- `50Hz`
- `10Hz`

> [!NOTE]

> > To achieve the full data rate as set by the `data_rate` value, it is highly advised to set the `update_interval` to zero.

Reviewer: ```suggestion
> To achieve the maximum data rate, configure `drdy_pin` and set `update_interval` to zero.
```
